packages/Search/src/details/DTK_DetailsUtils.hpp

```diff
@@ -252,6 +252,39 @@ typename View::non_const_type cloneWithoutInitializingNorCopying( View &v )
         Kokkos::ViewAllocateWithoutInitializing( v.label() ), v.layout() );
 }
 
+template <typename Value, typename View>
+inline View
+cast( View const &src,
+      typename std::enable_if<
+          std::is_same<typename View::value_type, Value>::value>::type * = 0 )
+{
+    return src;
+}
+
+template <typename Value, typename View>
+inline Kokkos::View<Value *, typename View::traits::array_layout,
+                    typename View::traits::device_type,
+                    typename View::traits::memory_traits>
+cast( View const &src,
```

Comment:
is `cast` the best name for this function? If the value types are different you are allocating and doing a copy. Maybe `changeViewDataType` or something similar?

Reply:
I couldn't come up with anything better.  I don't like `changeViewDataType`.  Anyone has a better idea?

Reply:
`type_cast`?